src/site/filters/liquid.js

```diff
@@ -244,6 +244,76 @@ module.exports = function registerFilters() {
     return JSON.stringify(getDeepLinks(currentPath, linksArray));
   };
 
+  liquid.filters.findCurrentPathDepthRecursive = (linksArray, currentPath) => {
```

Comment:
in this method, i had to disabled a lot of the eslint because it seems like it does not like recursive functions - overwriting variables seemed to be its biggest pet-peeve

Reply:
We'd prefer not to disable these eslint rules. disabling `no-shadow`, `no-param-reassign` are especially problematic. LMK if you'd like some help refactoring this.